### PR TITLE
Fix  slider thumb focus issue by reverting to using index as key

### LINKED ISSUE #233: Slider Thumb is not retaining its focus when value changed via keyboard
Slider Thumb is not retaining its focus when value changed via keyboard.

#### Expected behavior
A thumb is focused user should be able to adjust value by pressing arrow key multiple times.

#### Reproduction
```
export const ProcessChange: Story<Partial<SliderProps>> = (args) => {
  const [sliderValues, setSliderValues] = React.useState([20, 40]);

  return (
    <Slider
      {...args}
      values={sliderValues}
      onChange={(values) => setSliderValues([...values])}
    />
  );
};

ProcessChange.args = {
  min: 0,
  max: 60,
  tickLabels: ['0', '20', '40', '60'],
  tooltipProps: (index, val) => {
    return { placement: 'right', content: `\$${val}.00` };
  },
};
```

##### Steps to reproduce
1. Use above code 
1. Set focus to first thumb
1. Press Arrow key and note that after the first key press focus is lost

#### Additional information
See proposed fix  in  https://github.com/iTwin/iTwinUI-react/pull/232

### PR DESCRIPTION
**Pull request by bsteinbk**
If user used keyboard to change thumb value and there was an onChange listener which was creating a new `values` array, then the thumb would lose focus as it was re-rendered in its new position.

This fixes it by changing the Thumb key to user-provided `id` (with `index` as fallback) instead of maintaining an internal ref count.

This also fixes the issue in the unit tests where it was required to keep looking up the Thumb element because a new unique key was generated anytime the values array was updated.

Closes #233

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] ~~Add component features demo in Storybook (different stories)~~
- [x] ~~Approve test images for new stories~~
- [ ] Add an entry to the changelog for any new components and changes
- [x] Add screenshots of the key elements of the component
- [x] Add any additional comments describing the features you've implemented

### CODE REVIEW

File: src/core/Slider/Slider.tsx [hunk 1]
```diff
@@ -439,9 +439,13 @@ export const Slider = React.forwardRef<HTMLDivElement, SliderProps>(
...
           {currentValues.map((thumbValue, index) => {
             const [minVal, maxVal] = getAllowableThumbRange(index);
+            const thisThumbProps = thumbProps?.(index) ?? {};
+            const key = thisThumbProps.id
+              ? thisThumbProps.id
+              : index.toString();
             return (
               <Thumb
-                key={index}
+                key={key}
```
Comment (mayank99) [inline src/core/Slider/Slider.tsx:448]: Does this not work?

```suggestion
                key={`${thumbProps?.(index)?.id ?? index}`}
```

Comment (bsteinbk) [reply src/core/Slider/Slider.tsx]: I like being able to see the value in debugger... but I updated to mostly use your suggestion. I see no reason to run `thumbProps?.(index)` twice so I still create `thisThumbProps` and use if for the `key` and for `thumbProps` .

Comment (mayank99) [reply src/core/Slider/Slider.tsx]: Well, if you're assigning `{}` anyway, then you don't need `?.`

So you can change it to `key={thisThumbProps.id ?? index}`

**Review by mayank99 — APPROVED** [review]
Thanks!

**Pull request merged**